### PR TITLE
feat: add common parsable types

### PR DESCRIPTION
**Pull request by ajatprabha**

### CODE REVIEW

File: xload/type/endpoint.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package xloadtype
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+)
+
+type Endpoint struct {
```
Comment (sonnes) [inline xload/type/endpoint.go:9]: Can we extend the URL from the standard lib instead of creating a new type

File: xload/type/doc.go [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// Package xloadtype contains commonly used types for working with xload.Loader.
+package xloadtype
```
Comment (sonnes) [inline xload/type/doc.go:2]: What do you think of making this a top level `types`. Then these can be useful outside of xload

Comment (ajatprabha) [reply xload/type/doc.go:2]: Don't have a strong opinion but I have not encountered any need outside value parsing for such types.

Comment (ajatprabha) [reply xload/type/doc.go:2]: Also, `types` would be a very common package name. Can do `xtooltypes` or similar but I do not see any immediate usage. We can maybe move these later, if needed?

File: xload/type/endpoint.go (hunk 1 above)
Comment (ajatprabha) [reply xload/type/endpoint.go:9]: Let me try that, however, do you see things like path, query params being used?
I can imagine path being useful.

Comment (ajatprabha) [reply xload/type/endpoint.go:11]: Decided to keep `Endpoint` as well, but also added an aliased type for `url.URL` type.

**Pull request merged**